File: frontend/src/app/utils/Rating.ts
import { CustomUser } from "./user";
import { Book } from "./book";

export class Rating {
    id: number;
    user: CustomUser;
    book: Book;
    rating: number;
  
    constructor(id: number, user: CustomUser, book: Book, rating: number) {
      this.id = id;
      this.user = user;
      this.book = book;
      this.rating = rating;
    }
  }

File: backend2/core/views/rating.py
from rest_framework import viewsets
from django_filters.rest_framework import DjangoFilterBackend
from ..models import Rating
from ..serializers import RatingSerializer

class RatingViewSet(viewsets.ModelViewSet):
    queryset = Rating.objects.all()
    serializer_class = RatingSerializer
    filter_backends = [DjangoFilterBackend]
    filterset_fields = ['user', 'book']